geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionDataStore.java

```diff
@@ -270,19 +270,15 @@ public boolean isPartitionedRegionReady(PartitionedRegion partitionedRegion, fin
     if (colocatedWithList.size() == 0) {
       return partitionedRegion.isInitialized();
     }
-    Iterator itr = colocatedWithList.iterator();
-    return areAllColocatedPartitionedRegionsReady(bucketId, itr);
+    return areAllColocatedPartitionedRegionsReady(bucketId, colocatedWithList);
   }
 
-  private boolean areAllColocatedPartitionedRegionsReady(int bucketId, Iterator itr) {
-    while (itr.hasNext()) {
-      PartitionedRegion colocatedChildRegion = (PartitionedRegion) itr.next();
-      boolean success = isColocatedPartitionedRegionInitialized(colocatedChildRegion, bucketId);
-      if (!success) {
-        return success;
-      }
-    }
-    return true;
+  private boolean areAllColocatedPartitionedRegionsReady(int bucketId,
+      List<PartitionedRegion> colocatedWithList) {
+    return !colocatedWithList.stream()
+        .filter(partitionedRegion -> isColocatedPartitionedRegionInitialized(partitionedRegion,
+            bucketId) == false)
+        .findFirst().isPresent();
```

Comment:
Try:
```
return colocatedWithList.stream()
    .allMatch(partitionedRegion -> isColocatedPartitionedRegionInitialized(partitionedRegion, bucketId));
```

Reply:
I vote for "allMatch". It is just right.